lib/at_cmd_parser/at_params.c

```diff
@@ -28,9 +28,10 @@ static void at_param_clear(struct at_param *param)
 {
 	__ASSERT(param != NULL, "Parameter cannot be NULL.");
 
-	if ((param->type == AT_PARAM_TYPE_STRING) ||
-	    (param->type == AT_PARAM_TYPE_ARRAY)) {
+	if (param->type == AT_PARAM_TYPE_STRING) {
 		k_free(param->value.str_val);
+	} else if (param->type == AT_PARAM_TYPE_ARRAY) {
```

Comment:
I do agree it's cleaner this way, but was it really a leak? `str_val` and `array_val` are both member of the same union, and `k_free` doesn't really care about the pointer type?

Reply:
You are right, indeed not a leak. Thanks for the review.